File: Stack/StackUsingLL.js
class Node {
    constructor(data) {
      this.data = data;
      this.next = null;
    }
  }
  
  class Stack {
    constructor() {
      this.head = null;
    }
  
    isEmpty() {
      return this.head === null;
    }
  
    push(data) {
      const newNode = new Node(data);
      if (this.isEmpty()) {
        this.head = newNode;
        return;
      }
      newNode.next = this.head;
      this.head = newNode;
      console.log(this.head);
    }
  
    peek() {
      if (this.isEmpty()) {
        console.log("Nothing to show");
        return;
      }
      console.log(this.head.data);
    }
  
    pop() {
      if (this.isEmpty()) {
        console.log("Stack is empty");
        return;
      }
      const data = this.head.data;
      this.head = this.head.next;
      console.log(`${data} is deleted`);
    }
    firstElement(){
        let newNode =this.head;
        while(newNode.next!==null){
            newNode =newNode.next;
        }
        console.log('firstElement is ',newNode.data);
    }
    pushAtTheBottom(data){
        let initialNode =this.head;
        while(initialNode.next!==null){
            initialNode =initialNode.next;
        }
        const newNode =new Node(data);
        initialNode.next=newNode;
    }
  }
  
  const s = new Stack();
  s.push(4);
  s.push(6);
  s.push(3);
  s.firstElement();
  s.pushAtTheBottom(1);
  s.firstElement();
  // s.pop();
  // s.peek();
  

File: Stack/StackUsingLL.java
class StackUsingLL{
    static Node head;
    static class Node{
        int data;
        Node next;
        public Node(int data){
            this.data=data;
            this.next=null;
        }
    }
     static class Stack{
        public static boolean isEmpty(){
        return head==null;
    }
    public static void push(int data){
        Node newNode = new Node(data);
        if(isEmpty()){
            head=newNode;
            return;
        }
        newNode.next=head;
        head=newNode;
        System.out.println(head.data);
    }
    public static void peek(){
        if(isEmpty()){
            System.out.println("Nothing to show");
            return;
        }
        System.out.println(head.data);
    }
    public static void pop(){
        int data=head.data;
        head=head.next;
        System.out.println(data+ " is deleted");
    }
    public void pushAtTheBottom(int data) {
            Node initialNode = head;
            while (initialNode.next != null) {
                initialNode = initialNode.next;
            }
            Node newNode = new Node(data);
            initialNode.next = newNode;
        }
    }
    
    public static void main(String args[]){
        Stack s= new Stack();
        s.push(4);
         s.push(6);
          s.push(3);
        //   s.pop();
        // s.peek();
    }
}